src/utils/gaa-test.js

```diff
@@ -1290,6 +1295,46 @@ describes.realWin('GaaGoogle3pSignInButton', {}, () => {
       );
     });
 
+    it('should open an authorizationUrl in the same window when redirectMode is true', async () => {
+      // Show button.
+      GaaGoogle3pSignInButton.show({
+        allowedOrigins,
+        authorizationUrl: GOOGLE_3P_AUTH_URL,
+        redirectMode: true,
+      });
+      clock.tick(100);
+      await tick(10);
+      // Click button.
+      self.document.getElementById(GOOGLE_3P_SIGN_IN_BUTTON_ID).click();
+      clock.tick(100);
+      await tick(10);
+
+      expect(self.open.getCalls()[0].args[0].toString()).to.contain(
+        GOOGLE_3P_AUTH_URL +
+          '?' +
+          REDIRECT_SOURCE_URL_PARAM +
+          '=' +
+          encodeURIComponent(location.origin)
+      );
+      expect(self.open.getCalls()[0].args[1]).to.equal('_parent');
+    });
+
+    it('should open an authorizationUrl in a new window when redirectMode is false', async () => {
+      // Show button.
+      GaaGoogle3pSignInButton.show({
+        allowedOrigins,
+        authorizationUrl: GOOGLE_3P_AUTH_URL,
+        redirectMode: false,
```

Comment:
I think better to test default behavior when not passing redirectMode to ensure default behavior stays as before.

Reply:
Done. Please take a look.